kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStatusTable.js

```diff
@@ -324,44 +337,65 @@ define([
                     jobArray.forEach((jobState) => updateJobStatusInTable(jobState));
                 },
             });
+
+            const batchId = jobManager.model.getItem('exec.jobState.job_id');
             const paramsRequired = [];
             const jobIdList = [];
             jobs.forEach((jobState) => {
-                jobIdList.push(jobState.job_id);
-                if (!jobManager.model.getItem(`exec.jobs.params.${jobState.job_id}`)) {
-                    paramsRequired.push(jobState.job_id);
+                if (!jobState.batch_job) {
+                    jobIdList.push(jobState.job_id);
+                    if (
+                        !jobManager.model.getItem(`exec.jobs.params.${jobState.job_id}`) &&
+                        jobState.job_id !== batchId
+                    ) {
+                        paramsRequired.push(jobState.job_id);
+                    }
                 }
             });
+            jobManager.addListener('job-status', [batchId].concat(jobIdList));
 
             if (paramsRequired.length) {
                 jobManager.addListener('job-info', paramsRequired, {
                     jobStatusTable_info: handleJobInfo,
                 });
-                bus.emit('request-job-info', {
-                    jobIdList: paramsRequired,
-                });
+                const jobInfoRequestParams =
+                    paramsRequired.length === jobIdList.length
+                        ? { batchId }
+                        : { jobIdList: paramsRequired };
+                bus.emit('request-job-info', jobInfoRequestParams);
             }
-
-            // TODO: add listeners when the bulk import cell is initialised
-            jobManager.addListener('job-status', jobIdList, {
-                jobStatusTable_status: handleJobStatus,
-            });
-            jobManager.addListener('job-does-not-exist', jobIdList, {
-                jobStatusTable_dne: handleJobDoesNotExist,
-            });
+            bus.emit('request-job-status', { batchId });
         }
 
         /**
          * Update the table with a new jobState object
          * @param {object} jobState
          */
         function updateJobStatusInTable(jobState) {
+            const rowIx = jobState.retry_parent || jobState.job_id;
+
+            if (!jobsByRetryParent[rowIx]) {
+                // irrelevant
+                return;
+            }
+
+            const jobIx = `${jobState.created || 0}__${jobState.job_id}`;
+            jobsByRetryParent[rowIx].jobs[jobIx] = jobState;
+
+            // make sure that this job is the most recent version
+            const sortedJobs = Object.keys(jobsByRetryParent[rowIx].jobs).sort();
+            const lastJob = sortedJobs[sortedJobs.length - 1];
+            const jobUpdateData = jobsByRetryParent[rowIx].jobs[lastJob];
```

Comment:
a slightly horrible way to do this, but it sorts the children of a retry parent by their creation date, and selects the most recently created job for display

Reply:
Seems the most reasonable of the options.